### PR TITLE
Mes 551 code pre test data capture

### PR DESCRIPTION
**Pull request by zielinskikamil**
screen shot 2018-04-10 at 12 40 51

screen shot 2018-04-11 at 15 37 06

screen shot 2018-04-11 at 15 37 20

Added pre-test data capture page.
As you can see in screenshot no 2 it doesn't look good. I have a discussion with Neil about it. He said that this screen will be revisited soon and I can leave it like this.
I added one reusable modal to components (Modal for TellMeQuestions) and one modal to pages for recoding eyesight testing
I extracted select-element to the component.
I also checked why tslint extension for VSC does not warn us about unused variables. And it's because in the new version of tslint they stopped supporting this :(. 
The suggested workaround it turns similar flags in tsconfig but now they flag unused properties in types classes for our Cordova plugins. I have to prefix them with _. 
 
I don't know why I have so many changes in my PR :P - any suggestion why is that?

### CODE REVIEW

File: src/types/asam/index.ts [hunk 1]
```diff
@@ -29,15 +31,13 @@ import { Plugin, Cordova, IonicNativePlugin } from '@ionic-native/core';
...
 export class ASAM extends IonicNativePlugin {
-
   /**
    * Tries to toggle ASAM.
    * @param flag {boolean} true enables ASAM, false disable ASAM
    * @return {Promise<boolean>} Returns a promise that resolves with boolean value that describes if toggle succeeded
    */
   @Cordova()
-  toggle(flag: boolean): Promise<boolean> {
+  toggle(_flag: boolean): Promise<boolean> {
```
Comment (mathieuhendey) [inline src/types/asam/index.ts]: What's the reason behind prepending the `_` to `flag`? Also, the JSDoc block needs to be updated with the underscore

File: src/types/device-authentication/index.ts [hunk 2]
```diff
@@ -31,16 +33,14 @@ import { Plugin, Cordova, IonicNativePlugin } from '@ionic-native/core';
...
-
   /**
    * Run device authentication prompt. If FaceId/TouchId is set it shows prompt to use either one of them. If none of them are set it show passcode screen.
    * @param message {string} Message that is show in authentication prompt
    * @return {Promise<boolean>} Returns a promise that resolves when user tried to authenticate
    * @throws {Promise<string>} Returns a promise that rejects when there are some errors. Returned string contains localized message
    */
   @Cordova()
-  runAuthentication(message: string): Promise<boolean> {
+  runAuthentication(_message: string): Promise<boolean> {
```
Comment (mathieuhendey) [inline src/types/device-authentication/index.ts]: Again, why the underscore?

File: src/types/asam/index.ts [hunk 3]
```diff
@@ -1,3 +1,5 @@
+/* tslint:disable */
```
Comment (mathieuhendey) [inline src/types/asam/index.ts]: Why is this necessary? If it's unavoidable, can we disable the exact rule which is causing problems, rather than disabling TSLint entirely?

File: src/types/device-authentication/index.ts [hunk 4]
```diff
@@ -1,3 +1,5 @@
+/* tslint:disable */
```
Comment (mathieuhendey) [inline src/types/device-authentication/index.ts]: Same thing with disabling TSLint, can you disable the rule(s) which are giving you trouble rather than not linting this file at all?

File: src/components/select-button/select-button.ts [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import { Component, Input } from '@angular/core';
+
+/**
+ * Generated class for the SelectButtonComponent component.
+ *
+ * See https://angular.io/api/core/Component for more info on Angular
+ * Components.
+ */
```
Comment (danielchildsbjss) [inline src/components/select-button/select-button.ts]: Can we remove the generated text?

File: src/components/tell-me-modal/tell-me-modal.ts [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+ */
+@Component({
+  selector: 'tell-me-modal',
+  templateUrl: 'tell-me-modal.html'
+})
+export class TellMeModalComponent {
+  selectedTellMeQuestionId: string;
+
+  tellMeQuestions = [
+    { id: 'T1', today: 2, last7days: 8 },
```
Comment (danielchildsbjss) [inline src/components/tell-me-modal/tell-me-modal.ts]: Can we use real question text instead of numbers? We don't want the examiners to go out with their show me/tell me cards?

File: src/pages/eyesight-fault-recording-modal/eyesight-fault-recording-modal.ts [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+
+@Component({
+  selector: 'page-eyesight-fault-recording-modal',
+  templateUrl: 'eyesight-fault-recording-modal.html'
+})
+export class EyesightFaultRecordingModalPage {
+  signaturePadOptions: any;
+  @ViewChild(SignaturePad) signaturePad: SignaturePad;
+
+  constructor(public viewController: ViewController, configService: AppConfigProvider) {
```
Comment (danielchildsbjss) [inline src/pages/eyesight-fault-recording-modal/eyesight-fault-recording-modal.ts]: these can be private

File: src/types/asam/index.ts (hunk 1 above)
Comment (danielchildsbjss) [reply src/types/asam/index.ts]: Agree, avoid underscores if we can.

File: src/pages/pretest-checks/pretest-checks.ts [hunk 8]
```diff
@@ -1,25 +1,53 @@
...
+        this.tellMeQuestionId = selectedTellMeQuestionId;
+      }
+    });
 
-  ionViewDidLoad() {
-    console.log('ionViewDidLoad PretestChecksPage');
+    tellMeQuestionModal.present();
   }
 
+  showEyesightFaultRecordingModal() {
```
Comment (danielchildsbjss) [inline src/pages/pretest-checks/pretest-checks.ts]: I think this will eventually move to the abandon test screen (and pop up based on why you press abandon), we can leave it here for now though.

File: src/components/select-button/select-button.scss [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+      float: left;
+    }
+    .black-text {
+      color: black;
+    }
+    .arrow {
+      float: right;
+      color: #747474;
+      padding-left: 2px;
+      background-color: #d0d0d0;
```
Comment (ataylorBJSS) [inline src/components/select-button/select-button.scss]: Is this a colour available in the global $colors?

File: src/types/asam/index.ts (hunk 1 above)
Comment (zielinskikamil) [reply src/types/asam/index.ts]: sorry guys, I added a note to PR summary but I thought that it can be written even clearer. 
So the _ is a way to mute tsc warnings. But going to the very beginning.
I noticed that VSC do not show warnings about unused properties. I took a look why is that and found out that since new (5) version of tslint is not supporting unused variable rule. (see FAQ https://marketplace.visualstudio.com/items?itemName=eg2.tslint) and the recommended workaround it turns **noUnusedLocals**, **noUnusedParameters** on in tsconfig.json

I did it, but now it started complaining (correctly!) that **flag** is not used. and the only way to mute it was added _ a prefix

Actually, there is another solution, move these types classes so separate repo and import them whenever we need...

File: src/types/device-authentication/index.ts (hunk 2 above)
Comment (zielinskikamil) [reply src/types/device-authentication/index.ts]: Hi Mathieu, its connected with https://github.com/dvsa/mobile-examiner-alpha/pull/55#discussion_r180983592 (can we continue thread there?)

File: src/types/asam/index.ts (hunk 3 above)
Comment (zielinskikamil) [reply src/types/asam/index.ts]: Hi Mathieu, its connected with https://github.com/dvsa/mobile-examiner-alpha/pull/55#discussion_r180983592 (can we continue thread there?)

File: src/types/device-authentication/index.ts (hunk 4 above)
Comment (zielinskikamil) [reply src/types/device-authentication/index.ts]: Hi Mathieu, its connected with https://github.com/dvsa/mobile-examiner-alpha/pull/55#discussion_r180983592 (can we continue thread there?)

File: src/components/tell-me-modal/tell-me-modal.ts (hunk 6 above)
Comment (zielinskikamil) [reply src/components/tell-me-modal/tell-me-modal.ts]: hey Daniel, I implement it according to the designs and I got your point and I have asked the similar thing to Neil. He said that this screen will be revisited and I can leave it like this.

File: src/pages/pretest-checks/pretest-checks.ts (hunk 8 above)
Comment (zielinskikamil) [reply src/pages/pretest-checks/pretest-checks.ts]: the DE might first record the eyesight fault and then eg car registration number. To end test there is a special button at the button, but its good that you are thinking about such things !
